controllers/aws_creds.go

```diff
@@ -127,7 +127,12 @@ func (r *Saml2AwsReconciler) createAWSCreds(saml *samletv1.Saml2Aws) (*awsconfig
 	}
 	user, password := getLoginData(loginSecret)
 
-	account, err := formatAccount(r.Config, user, saml.Spec.RoleARN)
+	duration := r.Config.SessionDuration
+	if saml.Spec.SessionDuration != "" {
+		duration = saml.Spec.SessionDuration
```

Comment:
Makes sense to test that this config override by spec actually happens.

Reply:
on it

Reply:
test added